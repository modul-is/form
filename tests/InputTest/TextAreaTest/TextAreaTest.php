<?php

declare(strict_types=1);

require_once __DIR__ . '/../../bootstrap.php';

use Tester\Assert;

class TextAreaTest extends TestCase
{
	public function testRender()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Area');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/basic.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderPrepend()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Area')
			->setPrepend('prepend');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/prepend.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderAppend()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Area')
			->setAppend('append');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/append.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderIcon()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Area')
			->setIcon('user');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/icon.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderInputFloatingLabel()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Area')
			->setFloatingLabel(true);

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/floatingLabel.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderFormFloatingLabel()
	{
		$form = $this->getForm();

		$form->setFloatingLabel(true);

		$form->addTextArea('text', 'Area');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/floatingLabel.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderFormFloatingLabelInputDisable()
	{
		$form = $this->getForm();

		$form->setFloatingLabel(true);

		$form->addTextArea('text', 'Area')
			->setFloatingLabel(false);

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/basic.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderOptionId()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Area')
			->setOption('id', 'customId');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/id.latte'));

		Assert::same($html, $form->getComponent('text')->render()->__toString());
	}


	public function testRenderCustomTemplate()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Text')
			->setTemplate(__DIR__ . '/custom.latte');

		$string = 'custom-template';

		Assert::same($string, $form->getComponent('text')->render());
	}


	public function testRenderHidden()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Text')
			->setOption('hide', true);

		$string = '';

		Assert::same($string, $form->getComponent('text')->render());
	}


	public function testRenderSkip()
	{
		$form = $this->getForm();

		$form->addTextArea('text', 'Text')
			->setAutoRenderSkip();

		$string = '';

		Assert::same($string, $form->getComponent('text')->render());
	}
}

$testcase = new TextAreaTest;
$testcase->run();
