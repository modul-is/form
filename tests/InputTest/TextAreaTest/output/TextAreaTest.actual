<div class="form-group row"><div class="col-sm-4 control-label align-self-center"><label for="frm-text" class="col-form-label ">Area</label></div><div class="col-sm-8"><div class="input-group"><textarea name="text" id="frm-text" class="form-control  "></textarea><div class="input-group-append"><span class="input-group-text">append</span></div></div></div></div>