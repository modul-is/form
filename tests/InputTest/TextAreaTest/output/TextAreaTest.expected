<div class="form-group row" id="customId"><div class="col-sm-4 control-label align-self-center"><label for="frm-text" class="col-form-label ">Area</label></div><div class="col-sm-8"><div class="input-group"><textarea name="text" id="frm-text" class="form-control  "></textarea></div></div></div>